Automattic/wpcom.js
# Browser Compatibility: follow standard practices for isomorphic javascript

We can reduce complexity in all of our webpack, browserify, and rollup dependents by adding hints to `package.json` about how to avoid nodejs libs like `fs` when bundling for a browser.

I followed the instructions in [this blog post](https://nolanlawson.com/2017/01/09/how-to-write-a-javascript-package-for-both-node-and-the-browser/).

You can also find that other popular libraries like `superagent` use the [same technique](https://github.com/visionmedia/superagent/blob/master/package.json#L53).

**what changed**
There was a single function within `lib.site.media` that utilized `fs`. I extracted that function to its own file and made two versions: one with fs and one without.  Then by giving the `package.json` some hints about which to pick, upstream dependents can use this lib cleanly.

**to test**
- you can go to https://github.com/Automattic/wp-calypso/pull/22630 and verify that calypso builds even though i've removed the `fs` mock.
